architecture-building-systems/CityEnergyAnalyst
# Run code coverage on CEA test

This PR resolves #2715 by running coverage on `cea test` with the slow workflow. 
**This would require the user to have `coverage` installed for this to work.
Currently we have **73%** code coverage, meaning that 73% of our code in the slow workflow is being run without any errors.

Testing:
- Make sure `coverage` is installed i.e. `pip install coverage`
- Run `cea test --workflow slow` in the CEA console (might take a while)
- Should show coverage report after test finishes
- A coverage report would also be created in the indicated path and can be opened with the `index.html`


This also fixes #2730 by cleaning up schemas.yml and implementing checks for consistency of numerical ranges in unit test. Currently, the checks will only print the errors and not fail the unit test. There are still some values with 'TODO's especially for the weather file.

Testing:
- Run **main** for `cea.test.run_unit_tests`
- Could try changing some range values for numeric types and run the unit test again and it should show up in the output console
